docs/gallery/External_Resources.py

```diff
@@ -0,0 +1,144 @@
+# Using the add_key and the get_key methods
+# ------------
+# Mentioned in the note aobve, the user needs to reference a key object
+# when referencing a key. Let's start by creating a new key.
+
+# The add_key method will create a new key
+er.add_key(key='two')
+
+# The get_key method will return the key object of the key_name.
+key_object = er.get_key(key_name='two')
+
+# This will return a pandas dataframe of the KeyTable
+er.keys.to_dataframe()
+
+###############################################################################
+# Using the add_keys method
+# ------------
+# This is another wrapper function, but allows the use of a pandas DataFrame
+# to add/reference keys, resources, and entities.
+
+new_data = {
+    'key_name': 'two',
+    'resources_idx': 1,
+    'entity_id': '12345',
+    'entity_uri': 'entity_uri4'}
+
+df = pd.DataFrame(new_data, index=[0])
+
+er.add_keys(df)
+
+###############################################################################
+# Using the get_keys method
+# ------------
+# This method returns a DataFrame of key_name, resource_table_idx, entity_id,
+# and entity_uri. You can either have a single key object,
+# a list of key objects, or leave the input paramters empty to return all.
+
+# Single Key
+er.get_keys(keys=er.get_key('one'))
+
+# List of Specific Keys
+er.get_keys(keys=[er.get_key('one'), er.get_key('two')])
+
+# All Keys
+er.get_keys()
+
+###############################################################################
+# Using the add_resource method
+# ------------
+# This method creates new resource objects and adds them to the ResourceTable.
+# The ResourceTable holds the name and uri for the resource.
+# For example, you could add NCBI Taxonomy with its uri.
+
+er.add_resource(resource='taxonomy', uri='uri')
+
+###############################################################################
+# Using the add_entity method
+# ------------
+# This method creates a new entity object and adds it to the EntityTable.
+# Keeping with the NCBI Taxonomy example, the entity would be a specfic item
+# or "search" in NCBI Taxonomy, whereas the resource is NCBI Taxonomy itself.
+
+er.add_entity(key='four', resources_idx=1,
+              entity_id='113', entity_uri='entity_uri3')
+
+###############################################################################
+# Using the add_object method
+# ------------
+# This method creates a new object and adds it to the ObjectTable.
+
+er.add_object(container=data, field='')
+
+###############################################################################
+# Using the add_external_reference method
+# ------------
+# This method creates objectkey to ObjectKeyTable
+
+# This method below returns an Object object.
+object_ = er._check_object_field(data, field="")
+
+key_object = er.get_key(key_name='one')
+
+er.add_external_reference(object_, key_object)
+
+# This return ObjectKeyTable
+er.object_keys.to_dataframe()
```

Comment:
This is a good start! I would suggest tweaking the wording to be more user focused -- what do these fields represent in the real world, how are they related to each other, and when would a user want to use ExternalResources. If you can also use real-world examples -- `Homo sapien` is good, but please use the IDs and URIs from the NCBI species ontology

Reply:
This will be updated in #539.